### ISSUE TITLE
Mir does not compile on Debian Buster and capnproto 0.7.0-1

### ISSUE THREAD

**Issue by n3rdopolis** [MirServer/mir#680]
Hi

I am trying to compile Mir on Debian Testing. seems as of recent they switched to capnproto 0.7.0-1, and it's doing some odd stuff with cmake, and how it is finding the binary:

Scanning dependencies of target mirudev
Scanning dependencies of target mircookie
/bin/sh: 1: /usr/lib/bin/capnp: not found

I tried 
sed -i 's|/usr/lib/bin/capnp|/usr/bin/capnp|g' src/capnproto/CMakeFiles/mircapnproto.dir/build.make 
as a workaround, but I got more errors about a missing include

Thanks

**Comment by n3rdopolis** [comment]
I am using Mir Master. Maybe something's different in the way Debian ships it.

These are the arguments I pass to CMake
cmake -Wno-dev . -DCMAKE_INSTALL_PREFIX=$INSTALLDIR -DCMAKE_PREFIX_PATH=$INSTALLDIR -DCMAKE_BUILD_TYPE=MinSizeRel -DLIB_INSTALL_DIR=$INSTALLDIR/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH) -DMIR_ENABLE_TESTS=OFF -DMIR_BUILD_PLATFORM_EGLSTREAM_KMS=OFF

**Comment by Conan-Kudo** [comment]
@n3rdopolis No, the difference is in `capnproto`. Debian's capnproto is built with Autotools, so some stuff is just flat-out broken with its generated CMake stuff.

Fedora's `capnproto` (which I'm the maintainer of) is built with CMake: https://src.fedoraproject.org/rpms/capnproto/blob/master/f/capnproto.spec

And that means all the CMake module files are generated and installed properly.

**Comment by n3rdopolis** [comment]
Does this mean this has to be reported to Debian?

**Comment by Conan-Kudo** [comment]
@n3rdopolis Yep, most likely.

**Comment by tsimonq2** [comment]
I can confirm this; it's now blocking Mir from being uploaded to Debian.

**Comment by AlanGriffiths** [comment]
@tsimonq2 @n3rdopolis has anyone raised this ("some stuff is just flat-out broken with its generated CMake stuff") with Debian?

**Comment by tsimonq2** [comment]
Here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=920230

**Comment by tsimonq2** [comment]
A fix has been uploaded to Sid.

**Comment by kentonv** [comment]
Hi, Cap'n Proto maintainer here. FWIW, Cap'n Proto's own distro packages should be built with autotools, not cmake. We don't officially support the cmake build (of Cap'n Proto itself); it is provided "as-is" and maintained by contributors outside the core team. We do intend to support dependents using cmake. It sounds like there is a bug in the way the autotools build generates installed cmake import files used for this. We should fix that upstream, not just in a Debian-specific patch -- I'm happy to accept a PR.
